### ISSUE TITLE
Crashing with EXC_BAD_ACCESS when opening a buffer with Telugu Unicode text

### ISSUE THREAD

**Issue by sriharshachilakapati** [qvacua/vimr#722]
When I'm trying to open a JSON file with Telugu Unicode text in it, VimR immediately crashes. However this is not found in terminal NVIM.

Using VimR 0.24.0
Using NVIM v0.3.4

Attaching the crash file (changed extension to .log because GitHub doesn't allow otherwise).

The file I'm trying to open can be found here: https://github.com/sriharshachilakapati/yakyak/blob/master/src/locales/te.json

[VimR_2019-04-21-160553_Harshas-MacBook-Pro.crash.log](https://github.com/qvacua/vimr/files/3101117/VimR_2019-04-21-160553_Harshas-MacBook-Pro.log)

iTerm + TMUX + NVIM doesn't crash for me, but the text is appearing distorted.

**Comment by sriharshachilakapati** [comment]
Tried with updated VimR version. It's not crashing anymore, but not rendering fine too. Can we do something to support proportional fonts?

Output:

![VimR](https://user-images.githubusercontent.com/4593204/56469176-cb74bf80-6453-11e9-8827-40cadc11e7a6.png)

![NVIM TUI](https://user-images.githubusercontent.com/4593204/56469253-d0863e80-6454-11e9-951d-8db167a7d2ac.png)

Expected: (Rendered by Xcode)

![Xcode](https://user-images.githubusercontent.com/4593204/56469193-eba47e80-6453-11e9-8e29-aa0cb3d34ea9.png)

That character `<200c>` is a zero width joiner. It should not be rendered at all.

**Comment by qvacua** [comment]
@sriharshachilakapati Have you tried to turn on the "Use Ligatures" option in the Appearance preferences?

**Comment by sriharshachilakapati** [comment]
Yes @qvacua, I had ligatures enabled. I think the problem is that nvim and vimr both are not accepting the diacritics well. In Telugu and many other indic languages, diacritic characters can be added to the base character on top, bottom right of the base character.

Ligatures are important here, and I can see that the ligatures are rendered properly. Nothing wrong there. Only thing is that the advance widths are not taken into account, and hence I can see that the letters are overlapped on themselves.

There is nothing I can do to get it to work with TUI of nvim, because there I'm pretty much sure that the entire screen space will be divided into a character grid. So this is the reason I thought I can rely on the GUI.

Can you explain me how the text rendering is happening in VimR?
